extension/src/lib.rs

```diff
@@ -184,7 +184,9 @@ impl GapfillDeltaTransition {
         if self.window.back().map_or(false, |(prev, _)| *prev > time) {
             elog!(Error, "inputs must be in ascending time order")
         }
-        self.window.push_back((time, val));
+        if time >= self.current_window_min {
+            self.window.push_back((time, val));
+        }
```

Comment:
shouldn't this be a `elog(Error, ...)` in the aggregate functions?

Reply:
Not if you have a step of 30m and a range of 5m. Then every time you move windows you'll still have. 25 minutes of data to throw out